ffi/derive/src/ffi_fn.rs

```diff
@@ -61,21 +49,42 @@ pub fn gen_definition(fn_descriptor: &FnDescriptor) -> TokenStream {
     }
 }
 
-pub fn gen_fn_name(self_ty_name: Option<&Ident>, method_name: &Ident) -> Ident {
-    let self_ty_name = self_ty_name.map_or_else(Default::default, ToString::to_string);
+pub fn gen_fn_name(fn_descriptor: &FnDescriptor) -> Ident {
+    let method_name = &fn_descriptor.sig.ident;
+    let self_ty_name = fn_descriptor
+        .self_ty_name()
+        .map_or_else(Default::default, ToString::to_string);
+    let trait_name = fn_descriptor
+        .trait_name()
+        .map_or_else(Default::default, |trait_name| format!("__{trait_name}"));
 
     Ident::new(
-        &format!("{}__{}", self_ty_name, method_name),
+        &format!("{}{}__{}", self_ty_name, trait_name, method_name),
```

Comment:
```suggestion
        &format!("{}_{}__{}", self_ty_name, trait_name, method_name),
```
Imagine `NameFromStr__from_str` vs `Name_FromStr__from_str`